opentelemetry/src/lib.rs

```diff
@@ -55,40 +35,109 @@
 //! # }
 //! ```
 //!
+//! See the [examples](https://github.com/open-telemetry/opentelemetry-rust/tree/main/examples) directory for different integration patterns.
+//!
 //! See the [`trace`] module docs for more information on creating and managing
 //! spans.
 //!
 //! [`Span`]: crate::trace::Span
 //!
-//! # Metrics
-//!
-//!
-//! The [`metrics`] module includes types for recording measurements about a
-//! service at runtime.
-//!
-//! ### Creating instruments and recording measurements
-//!
+//! # Getting Started with Metrics
+//!
+//! The [`metrics`] module provides types for recording measurements about a
+//! service at runtime. Below are the key steps to report measurements using
+//! OpenTelemetry Metrics:
+//!
+//! 1. **Obtain a Meter:** Get a `Meter` from a `MeterProvider`.
+//! 2. **Create Instruments:** Use the `Meter` to create one or more instruments
+//!    (e.g., counters, histograms).
+//! 3. **Record Measurements:** Use the instruments to record measurement values
+//!    along with optional attributes.
+//!
+//! ## How Metrics Work in OpenTelemetry
+//! In OpenTelemetry, raw measurements recorded using instruments are
+//! **aggregated in memory** to form metrics. These aggregated metrics are
+//! periodically exported by the [`opentelemetry_sdk`] at fixed intervals (e.g.,
```

Comment:
Since this is a general introduction, should we talk about pull based exporters as well?

Reply:
in future iteration, yes.